packages/agents-openai/src/openaiResponsesModel.ts

```diff
@@ -512,86 +591,117 @@ function convertToOutputItem(
 ): protocol.OutputModelItem[] {
   return items.map((item) => {
     if (item.type === 'message') {
-      const { id, type, role, content, status, ...remainingItem } = item;
+      const { id, type, role, content, status, ...providerData } = item;
       return {
-        type,
         id,
+        type,
         role,
         content: content.map(convertToMessageContentItem),
         status,
-        providerData: remainingItem,
+        providerData: snakeToCamelCase(providerData),
       };
     } else if (
       item.type === 'file_search_call' ||
       item.type === 'web_search_call' ||
       item.type === 'image_generation_call' ||
       item.type === 'code_interpreter_call'
     ) {
-      const { id, type, status, ...remainingItem } = item;
-      const outputData =
-        'result' in remainingItem && remainingItem.result !== null
-          ? remainingItem.result // type: "image_generation_call"
-          : undefined;
+      const { status, ...remainingItem } = item;
+      let outputData = undefined;
+      if ('result' in remainingItem && remainingItem.result !== null) {
+        // type: "image_generation_call"
+        outputData = remainingItem.result;
+        delete (remainingItem as any).result;
+      }
       const output: protocol.HostedToolCallItem = {
         type: 'hosted_tool_call',
-        id,
```

Comment:
The id property could be used in many cases, so i would like to consistently have it also in providerData.